irods/resource.py

```diff
@@ -38,6 +38,54 @@ def __init__(self, manager, result=None):
 
         self._meta = None
 
+
+    ## Cached properties to expose parent id or name regardless whether the DB model is iRODS 4.1- or 4.2+
+
+    @property
+    def parent_id(self):
+        if self.parent == None:
+            return None
+        if not getattr(self,'_parent_id',''):
+            sess = self.manager.sess
+            if sess.server_version >= (4, 2, 0):
+                self._parent_id = self.parent
+            else:
+                self._parent_id = sess.query(Resource).filter(Resource.name == self.parent).one()[Resource.id]
+        return self._parent_id
+
+
+    @property
+    def parent_name(self):
+        if self.parent == None:
+            return None
+        if not getattr(self,'_parent_name',''):
+            sess = self.manager.sess
+            if sess.server_version < (4, 2, 0):
+                self._parent_name = self.parent
+            else:
+                self._parent_name = sess.query(Resource).filter(Resource.id == self.parent).one()[Resource.name]
+        return self._parent_name
+
+    ## Cached property to expose resource hierarchy string
+
+    @property
+    def resc_hierarchy(self):
```

Comment:
not just ... hierarchy?    this is a property of a resource... so already has that context?

Reply:
or `hierachy_string` maybe.

Reply:
agreed, it will just be `hierarchy`.

Reply:
Cannot resolve the conversation however, maybe because it's in draft status.